file-formats/aobj/type/zif_aff_aobj_v1.intf.abap

```diff
@@ -0,0 +1,69 @@
+INTERFACE zif_aff_aobj_v1
+  PUBLIC.
+
+  TYPES:
+    BEGIN OF ty_attributes,
+      "! <p class="shorttext">Write Class</p>
+      "! Name of executable class or program which writes the data to the archive
+      "! $required
+      write_class  TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Delete Class</p>
+      "! Name of executable class or program which deletes the data from the tables
+      "! $required
+      delete_class TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Reload Class</p>
+      "! Name of executable class or program which reloads the data to the tables (may be empty)
+      reload_class TYPE zif_aff_types_v1=>ty_object_name_30,
+    END OF ty_attributes.
+
+  TYPES:
+    "! <p class="shorttext">Table to Be Archived</p>
+    "! Tables of business object instance from which data is archived
+    BEGIN OF ty_table_to_be_archived,
+      "! <p class="shorttext">Name</p>
+      "! Name of table or structure
+      "! $required
+      name TYPE c LENGTH 30,
+    END OF ty_table_to_be_archived.
+
+  "! <p class="shorttext">Tables to Be Archived</p>
+  "! Tables from which data is archived
+  TYPES ty_tables_to_be_archived TYPE SORTED TABLE OF ty_table_to_be_archived WITH UNIQUE KEY name.
+
+  TYPES:
+    "! <p class="shorttext">Name of Archiving Class</p>
+    "! Archiving classes are no ABAP classes, but IDs of reuse objects, whose data should be archived
+    "! Together with the business object instance (for example change documents).
+    BEGIN OF ty_archiving_class,
+      "! <p class="shorttext">Name</p>
+      "! Name of archiving class
+      "! $required
+      name TYPE zif_aff_types_v1=>ty_object_name_30,
```

Comment:
Just to be sure. Are archiving classes 30 characters? Somehow I have in mind they are shorter.

Reply:
I changed it to 10 10 characters

Reply:
Thanks